### ISSUE TITLE
Use default property values when adding a new column in a migration

### ISSUE THREAD

**Issue by segiddins** [realm/realm-cocoa#1793]

**Comment by porsager** [comment]
This one bit me as well. Is there any timeframe on this or would it maybe be an idea to note it in the migration documentation?

**Comment by jpsim** [comment]
I just added this to our docs: https://realm.io/docs/objc/latest/#migrations

> Note that [default property values](https://realm.io/docs/objc/latest#default-property-values) aren't applied to new objects during migrations. We consider this to be a bug and are tracking it as [#1793](https://github.com/realm/realm-cocoa/issues/1793).

**Comment by rsahara** [comment]
I hope this will be fixed soon as it's difficult for the developers to test if the migrated DB works for each of the previous versions of the realm. (after 90 updates in our case)

We recently received errors because of unexpected null string properties (with normally a default value of @""), when we released a major update of our app.

**Comment by jpsim** [comment]
The challenge here is to only ever set default values if they haven't explicitly been set by a user, which in turn requires tracking which setters were invoked.

**Comment by JadenGeller** [comment]
This does seem challenging. I would imagine that a user explicitly set a properties value back to the default, it should not be automatically migrated if the default changes. It almost seems that there ought to be some sort of `resetProperty:` method.

**Comment by JadenGeller** [comment]
@jpsim Couldn't we also strictly set the default values _before_ the user sets the property?

**Comment by jpsim** [comment]
Yes, I think eagerly setting default values before user set the property would be acceptable. It _shouldn't_ have a significant performance impact, but I'd prefer if we checked nonetheless.

**Comment by jemshit** [comment]
So what is solution for assigning default value for new Column after migration ? any magical method ?

**Comment by werediver** [comment]
No magic. Set manually :(

**Comment by guhan0** [comment]
Any solution or fix to this issue?

**Comment by ThreadPitt** [comment]
Still happens here, any progress on this?
